### PR TITLE
Use `Session` in CLI

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
1. `Session` has now two APIs: for chaining (`with_*` and `*_and`) and singular actions
2. removed duplicated code from `drink-cli/executor` (core code was duplicating `Session`)
3. some mambo-jumbo for sharing `ContractMessageTranscoder` (it is not `Clone` nor `Copy`...)

### CODE REVIEW

File: drink/src/session.rs [hunk 1]
```diff
@@ -93,33 +124,58 @@ impl Session {
...
+    pub fn call_and(
+        mut self,
+        address: Option<AccountId32>,
+        message: &str,
+        args: &[String],
+    ) -> Result<Self, SessionError> {
+        self.call(address, message, args).map(|_| self)
+    }
+
+    pub fn call(
```
Comment (piotrMocz) [inline drink/src/session.rs]: This is not very important but actually for a public API I'd prefer two methods:
```rust
pub fn call(&mut self, message: &str, args: &[String])
```
and:
```rust
pub fn call_with_address(&mut self, address: AccountId32, message: &str, agrs: &[String])
```
And have them both call a non-public `call_internal` method with a signature just like `call` now.
Just thinking out loud :)

File: examples/flipper/lib.rs [hunk 2]
```diff
@@ -67,12 +67,12 @@ mod tests {
 
     #[test]
     fn flipping() -> Result<(), Box<dyn Error>> {
-        let init_value = Session::new(Some(transcoder()))?
-            .deploy(bytes(), "new", &["true"], vec![])?
-            .call("flip", &[])?
-            .call("flip", &[])?
-            .call("flip", &[])?
-            .call("get", &[])?
+        let init_value = Session::new(Some(Rc::new(transcoder())))?
```
Comment (piotrMocz) [inline examples/flipper/lib.rs]: This probably deserves its own wrapper to hide the gory details :)

**Review by piotrMocz — APPROVED** [review]
To the extent that I understand the code, the changes seem great.
I left some nitpicky comments but please feel free to ignore them :)

File: drink/src/session.rs (hunk 1 above)
Comment (pmikolajczyk41) [reply drink/src/session.rs]: done :ok_hand:

File: examples/flipper/lib.rs (hunk 2 above)
Comment (pmikolajczyk41) [reply examples/flipper/lib.rs]: I moved them to the `transcoder` function - I hope that it will be soon hidden by the ink e2e framework

**Pull request merged**